### PR TITLE
Recommend using PEP 585

### PR DESCRIPTION
**Pull request by srittau**
Also remove conventions enforced by black and remove old note about optional default arguments (now part of PEP 484 and enforced by CI).

Unfortunately, due to python/mypy#9980 we need a special rule for `Tuple[x, ...]`.

### CODE REVIEW

**Comment by srittau** [comment]
Weird that the pytype check fails here, while is succeeded in #5050.

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -241,7 +237,12 @@ checker, and leave out unnecessary detail:
 * use `float` instead of `Union[int, float]`.
 
 Some further tips for good type hints:
-* avoid invariant collection types (`List`, `Dict`) in argument
+* use built-in generics (`list`, `dict`, `tuple`, `set`), instead
+  of importing them from `typing`, **except** for arbitrary length tuples
+  (`Tuple[int, ...]`);
```
Comment (JelleZijlstra) [inline CONTRIBUTING.md:242]: It would be helpful here to link to the mypy issue, so we can easily refer back to it.

**Comment by JelleZijlstra** [comment]
The pytype failures are strange:
```
stdlib/random.pyi (3.6): TypeError: unhashable type: 'list'
stdlib/secrets.pyi (3.6): TypeError: unhashable type: 'list'
```
@rchen152 do you know what could be causing this?

**Comment by rchen152** [comment]
> The pytype failures are strange:
> 
> ```
> stdlib/random.pyi (3.6): TypeError: unhashable type: 'list'
> stdlib/secrets.pyi (3.6): TypeError: unhashable type: 'list'
> ```
> 
> @rchen152 do you know what could be causing this?

They're strange because pytype is crashing :/
```
Traceback (most recent call last):
  [...]
  File "/usr/local/google/home/rechen/.venv_py38/lib/python3.8/site-packages/pytype/pytd/pytd.py", line 468, in _FlattenTypes
    unique = tuple(collections.OrderedDict.fromkeys(flattened))
  File "<attrs generated hash pytype.pytd.pytd.GenericType>", line 3, in __hash__
TypeError: unhashable type: 'list'
```
Looking.

**Comment by rchen152** [comment]
This is fixed in pytype-2021.02.23. (I retriggered the pytype check, and it seems happy now.)

**Comment by JelleZijlstra** [comment]
Awesome, thanks for the quick fix!

**Pull request merged**